services/replicator/replicatorclientfactory.go

```diff
@@ -172,8 +172,8 @@ func (f *repCltFactoryImpl) GetHostsForAllDeployment() map[string][]string {
 func (f *repCltFactoryImpl) UpdateHostsForDeployment(deployment string, hosts []string) {
 	f.hostLk.Lock()
 	f.deploymentClientsLk.Lock()
-	defer f.deploymentClientsLk.Unlock()
 	defer f.hostLk.Unlock()
+	defer f.deploymentClientsLk.Unlock()
```

Comment:
Ideally, best to have the 'defer lk.Unlock()" immediately follow the line that does "lk.Lock()".